src/components/media-player/youtube-thumbnail-url-cache.js

```diff
@@ -0,0 +1,118 @@
+import Logger from '../../common/logger.js';
+import { isBrowser } from '../../ui/browser-helper.js';
+import { getDisposalInfo, DISPOSAL_PRIORITY } from '../../common/di.js';
+import { analyzeMediaUrl, verifyYoutubeThumbnailUrl } from '../../utils/media-utils.js';
+
+const logger = new Logger(import.meta.url);
+
+class YoutubeThumbnailUrlCache {
+  static get inject() { return []; }
+
+  constructor() {
+    this._isDisposed = false;
+    this._entries = new Map();
+    this._subscribers = new Set();
+  }
+
+  getEntry(sourceUrl) {
+    this._throwIfDisposed();
+    let entry = this._entries.get(sourceUrl);
+    if (!entry) {
+      const { isYoutube, youtubeVideoId } = analyzeMediaUrl(sourceUrl);
+      if (isYoutube && youtubeVideoId) {
+        const lowResUrl = `https://i.ytimg.com/vi/${encodeURIComponent(youtubeVideoId)}/hqdefault.jpg`;
+        const highResUrl = `https://i.ytimg.com/vi/${encodeURIComponent(youtubeVideoId)}/maxresdefault.jpg`;
+        entry = this._createUnverifiedYoutubeEntry(sourceUrl, youtubeVideoId, lowResUrl, highResUrl);
+        this._entries.set(sourceUrl, entry);
+        this._tryVerifyHighResThumbnailUrl(entry);
+      } else {
+        entry = this._createEphemeralNonYoutubeEntry(sourceUrl);
+      }
+    }
+
+    return entry;
+  }
+
+  subscribe(callback) {
+    this._throwIfDisposed();
+    this._subscribers.add(callback);
+  }
+
+  unsubscribe(callback) {
+    this._throwIfDisposed();
+    this._subscribers.delete(callback);
+  }
+
+  _createUnverifiedYoutubeEntry(sourceUrl, videoId, lowResThumbnailUrl, highResThumbnailUrl) {
+    return {
+      sourceUrl,
+      videoId,
+      lowResThumbnailUrl,
+      highResThumbnailUrl,
+      isHighResThumbnailUrlVerfied: false
+    };
+  }
+
+  _createEphemeralNonYoutubeEntry(sourceUrl) {
+    return {
+      sourceUrl,
+      videoId: null,
+      lowResThumbnailUrl: null,
+      highResThumbnailUrl: null,
+      isHighResThumbnailUrlVerfied: false
+    };
+  }
+
+  async _tryVerifyHighResThumbnailUrl(entry) {
+    if (!isBrowser()) {
+      return;
+    }
+
+    try {
+      const sucess = await verifyYoutubeThumbnailUrl(entry.highResThumbnailUrl);
```

Comment:
This guy needs another "c" to be truly successful 😋

Reply:
:grin: I helped the poor guy.